### PR TITLE
Remove old rosinstall and init submodule in cmake

### PR DESCRIPTION
**Pull request by mqcmd196**
We don't need to rebuild `cv_bridge` in melodic environment. so remove rosinstall.
Instead, the submodule is not initialized so update the git submodule when building it.

### CODE REVIEW

**Review by knorth55 — APPROVED** [review]

**Comment by mqcmd196** [comment]
BTW, who has the authority to merge PRs? (@knorth55 or @k-okada ??) I want maintainers to fix CIs and merge this PR for beginner users to install this package. If you don't have to do it, I'll handle it.

**Comment by k-okada** [comment]
@mqcmd196 The current Admins are @k-okada and @knorth55. But before merging, the PR needs to pass testing, and I don't think it is the maintainer's role alone to modify the PR so that it passes CI.

**Comment by knorth55** [comment]
https://github.com/jsk-ros-pkg/coral_usb_ros/blob/ff92ae3135e74d0e66c3d056754c389e1d987504/.github/workflows/main.yml#LL42C1-L46C27

submodule is checked out, so please change cmake something like this.

```cmake
find_package(Git REQUIRED)
if (NOT EXISTS ${PROJECT_SOURCE_DIR}/python/coral_usb/posenet/pose_engine.py)
  add_custom_target(posenet_update ALL
    COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive python/coral_usb/posenet
    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
  )
endif()
```

**Comment by mqcmd196** [comment]
thank you. all test passed

**Comment by mqcmd196** [comment]
@k-okada Kindly ping. CI green

**Comment by k-okada** [comment]
1) OK: `We don't need to rebuild cv_bridge in melodic environment.` 
 -> https://github.com/jsk-ros-pkg/coral_usb_ros/pull/92 uses cv_bridge_python3 so that we do not need compile cv_bridge. 
2) ??: `so remove rosinstall.` are you sure about that ? There are some code using `fc.rosinstall` https://github.com/search?q=repo%3Ajsk-ros-pkg%2Fcoral_usb_ros%20fc.rosinstall&type=code
https://github.com/actions/checkout/blob/8530928916aaef40f59e6f221989ccb31f5759e7/README.md?plain=1#L107-L114
3) ?? `submodule is checked out, so please change cmake something like this.` wstool update checkout submodules, so why we need to call git submodule update?` Basicall, I am reluctant with checkout submodule within CMake. Is this a common thing these days? https://github.com/search?q=ros+path%3A%22.gitmodules%22&type=code

**Comment by mqcmd196** [comment]
> ??: so remove rosinstall. are you sure about that ? There are some code using fc.rosinstall https://github.com/search?q=repo%3Ajsk-ros-pkg%2Fcoral_usb_ros%20fc.rosinstall&type=code
> https://github.com/actions/checkout/blob/8530928916aaef40f59e6f221989ccb31f5759e7/README.md?plain=1#L107-L114

I found `fc.rosinstall` used in Dockerfile. I fixed the Dockerfile and confirmed that it was built correctly. https://github.com/jsk-ros-pkg/coral_usb_ros/pull/132/commits/ea22c826ca16f7e7b536785706673ab89d3ebcd2

> ?? submodule is checked out, so please change cmake something like this. wstool update checkout submodules, so why we need to call git submodule update?` 

IMO, when I install a single ROS repository, I prefer to clone it under src and build it as-is rather than using wstool. While wstool is useful for managing multiple repositories simultaneously, I think it is best not to use it if you don't have to. We can omit the step. 

> Basicall, I am reluctant with checkout submodules within CMake. Is this a common thing these days? https://github.com/search?q=ros+path%3A%22.gitmodules%22&type=code

Hmm, I'm not sure...

File: docker/Dockerfile [hunk 1]
```diff
@@ -67,26 +67,19 @@ ARG ROS_DISTRO
...
       pip install dlib==19.21.1; \
       pip install fcn chainercv chainer==6.7.0 protobuf==3.18.0; \
     fi
 
-# /opt/ros/${ROS_DISTRO}/share can be changed after rosdep install, so we run it 3 times.
 RUN rosdep update --include-eol-distros && \
     apt-get update && \
-    for i in $(seq 3); do \
-      rosdep install --rosdistro ${ROS_DISTRO} -r -y -i --from-paths /opt/ros/${ROS_DISTRO}/share ~/ros/${ROS_DISTRO}/src; \
-    done && \
```
Comment (mqcmd196) [inline docker/Dockerfile:89]: Before my PR, this line doesn't work in kinetic docker
```
213.6 ERROR: Rosdep experienced an internal error.
213.6 Please go to the rosdep page [1] and file a bug report with the message below.
213.6 [1] : http://www.ros.org/wiki/rosdep
213.6
213.6 rosdep version: 0.21.0
213.6
213.6 Traceback (most recent call last):
213.6   File "/usr/lib/python2.7/dist-packages/rosdep2/lookup.py", line 432, in resolve_all
213.6     resolutions_flat = depend_graph.get_ordered_dependency_list()
213.6   File "/usr/lib/python2.7/dist-packages/rosdep2/dependency_graph.py", line 114, in get_ordered_dependency_list
213.6     self.validate()
213.6   File "/usr/lib/python2.7/dist-packages/rosdep2/dependency_graph.py", line 96, in validate
213.6     % dependency)
213.6 KeyError: 'Invalid Graph Structure: rosdep key `python-numpy` does not exist in the dictionary of resolutions.'
213.6
213.6
215.1
215.1 ERROR: Rosdep experienced an internal error.
215.1 Please go to the rosdep page [1] and file a bug report with the message below.
215.1 [1] : http://www.ros.org/wiki/rosdep
215.1
215.1 rosdep version: 0.21.0
215.1
215.1 Traceback (most recent call last):
215.1   File "/usr/lib/python2.7/dist-packages/rosdep2/lookup.py", line 432, in resolve_all
215.1     resolutions_flat = depend_graph.get_ordered_dependency_list()
215.1   File "/usr/lib/python2.7/dist-packages/rosdep2/dependency_graph.py", line 114, in get_ordered_dependency_list
215.1     self.validate()
215.1   File "/usr/lib/python2.7/dist-packages/rosdep2/dependency_graph.py", line 96, in validate
215.1     % dependency)
215.1 KeyError: 'Invalid Graph Structure: rosdep key `python-numpy` does not exist in the dictionary of resolutions.'
215.1
215.1
------
Dockerfile:85
--------------------
  84 |     # /opt/ros/${ROS_DISTRO}/share can be changed after rosdep install, so we run it 3 times.
  85 | >>> RUN rosdep update --include-eol-distros && \
  86 | >>>     apt-get update && \
  87 | >>>     for i in $(seq 3); do \
  88 | >>>       rosdep install --rosdistro ${ROS_DISTRO} -r -y -i --from-paths /opt/ros/${ROS_DISTRO}/share ~/ros/${ROS_DISTRO}/src; \
  89 | >>>     done && \
  90 | >>>     rm -rf /var/lib/apt/lists/*
  91 |
--------------------
```

File: docker/Dockerfile [hunk 2]
```diff
@@ -67,26 +67,19 @@ ARG ROS_DISTRO
...
+    rosdep install --rosdistro ${ROS_DISTRO} -r -y -i --from-paths /opt/ros/${ROS_DISTRO}/share ~/ros/${ROS_DISTRO}/src || true && \
```
Comment (mqcmd196) [inline docker/Dockerfile:82]: Without this line, `rosdep` command returns 1 for not finding the dependencies. So force returning 0 by `|| true`

**Pull request merged**